crates/biome_js_analyze/src/analyzers/nursery/no_useless_lone_block_statements.rs

```diff
@@ -0,0 +1,168 @@
+use crate::semantic_services::Semantic;
+use crate::JsRuleAction;
+use biome_analyze::{context::RuleContext, declare_rule, ActionCategory, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make;
+use biome_js_syntax::{
+    AnyJsStatement, JsBlockStatement, JsFileSource, JsLabeledStatement, JsStatementList,
+    JsVariableStatement, TextRange,
+};
+use biome_rowan::{AstNode, AstNodeList, BatchMutationExt};
+
+declare_rule! {
+    /// Disallow unnecessary nested block statements.
+    ///
+    /// > In JavaScript, prior to ES6, standalone code blocks delimited by curly braces do not create a new scope and have no use.
+    /// > In ES6, code blocks may create a new scope if a block-level binding (let and const), a class declaration or a function declaration (in strict mode) are present. A block is not considered redundant in these cases.
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/no-lone-blocks
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// {}
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// if (foo) {
+    ///   bar();
+    ///   {
+    ///     baz();
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// while (foo) {
+    ///   bar();
+    /// }
+    /// ```
+    ///
+    pub(crate) NoUselessLoneBlockStatements {
+        version: "next",
+        name: "noUselessLoneBlockStatements",
+        recommended: false,
+    }
+}
+
+impl Rule for NoUselessLoneBlockStatements {
+    type Query = Semantic<JsBlockStatement>;
+    type State = TextRange;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let block = ctx.query();
+        let is_module = ctx.source_type::<JsFileSource>().is_module();
+
+        if let Some(_labeled_statement) = JsLabeledStatement::cast(block.syntax().parent()?) {
```

Comment:
You can use this to avoid an unnecessary cast:
```suggestion
        if JsLabeledStatement::can_cast(block.syntax().parent()?.kind()) {
```

Reply:
Lovely, have made this change